app/lib/features/invite_members/pages/invite_space_members.dart

```diff
@@ -0,0 +1,191 @@
+          _buildDoneButton(),
+          const SizedBox(height: 10),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildParentSpace() {
+    final parentSpace =
+        ref.watch(canonicalParentProvider(widget.roomId)).valueOrNull;
+
+    if (parentSpace == null) return const SizedBox.shrink();
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(L10n.of(context).parentSpace),
+        SpaceMemberInviteCard(
+          space: parentSpace.space,
+          isSelected: selectedSpaces.contains(parentSpace.space),
+          onChanged: (value) {
+            if (selectedSpaces.contains(parentSpace.space)) {
+              selectedSpaces.remove(parentSpace.space);
+            } else {
+              selectedSpaces.add(parentSpace.space);
+            }
+            setState(() {});
+          },
+        ),
+        const SizedBox(height: 16),
+        Text(L10n.of(context).otherSpaces),
+      ],
+    );
+  }
+
+  Widget _buildOtherSpace() {
+    final otherSpaces =
+        ref.watch(otherSpacesForInviteMembersProvider(widget.roomId));
+    return otherSpaces.when(
+      data: _buildOtherSpaceData,
+      error: (error, stack) => ListTile(
+        title: Text(error.toString()),
+      ),
+      loading: () => _buildSkeletonizerLoading(),
+    );
+  }
+
+  Widget _buildOtherSpaceData(List<Space> data) {
+    return Expanded(
+      child: ListView.builder(
+        itemCount: data.length,
+        shrinkWrap: true,
+        itemBuilder: (context, index) {
+          final space = data[index];
+          return SpaceMemberInviteCard(
+            space: space,
+            isSelected: selectedSpaces.contains(space),
+            onChanged: (value) {
+              if (selectedSpaces.contains(space)) {
+                selectedSpaces.remove(space);
+              } else {
+                selectedSpaces.add(space);
+              }
+              setState(() {});
+            },
+          );
+        },
+      ),
+    );
+  }
+
+  Widget _buildSkeletonizerLoading() {
+    return Skeletonizer(
+      child: ListView(
+        shrinkWrap: true,
+        children: [
+          ListTile(title: Text(L10n.of(context).loading)),
+          ListTile(title: Text(L10n.of(context).loading)),
+          ListTile(title: Text(L10n.of(context).loading)),
+          ListTile(title: Text(L10n.of(context).loading)),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildDoneButton() {
+    return ActerPrimaryActionButton(
+      onPressed: _inviteMembers,
+      child: Text(L10n.of(context).invite),
+    );
+  }
+
+  Future<void> _inviteMembers() async {
+    if (selectedSpaces.isEmpty) {
+      EasyLoading.showToast(L10n.of(context).pleaseSelectSpace);
+      return;
+    }
+
+    EasyLoading.show(
+      status: L10n.of(context).invitingSpaceMembersLoading,
+      dismissOnTap: false,
+    );
+
+    try {
+      final currentSpace = ref.watch(spaceProvider(widget.roomId)).valueOrNull;
```

Comment:
I know it's a bit confusing, but for one time call function, like actions or on-tap or something, these should be `ref.read` not `watch`...

Reply:
Not confusing, it is my mistake. Need to `ref.read` on any function call for single time read operation where `ref. watch` need to use in widget tree for updating state.

Let me do needful changes here.